src/command_modules/azure-cli-network/azure/cli/command_modules/network/_validators.py

```diff
@@ -1280,6 +1283,72 @@ def validate_custom_error_pages(namespace):
     namespace.custom_error_pages = values
 
 
+def validate_custom_headers(namespace):
+
+    if not namespace.monitor_custom_headers:
+        return
+
+    values = []
+    for item in namespace.monitor_custom_headers:
+        try:
+            item_split = item.split('=', 1)
+            values.append({'name': item_split[0], 'value': item_split[1]})
```

Comment:
If values are allowed to have escaped equals signs, then we'll need a more robust split strategy here. If it's always a base64 encoded or similar, then we're okay.

Reply:
This is the same strategy used that has always been used with `--tags` and we've not had complaints. I believe it is because of the simpler constraints on the key. If there's an equal sign in the value (escaped or not) it shouldn't be an issue.

Reply:
Err, so I may not need to worry about this. If the headers have already been correctly URL encoded, any equal sign should just show up as `%3D`. I imagine we're okay requiring properly encoded info here.

Reply:
(sorry, looks like there was a race condition, and I didn't see your message when I was typing mine.)